tests/integration/test_scenario_1.py

```diff
@@ -1,137 +1,156 @@
-#         assert span_server.attributes["SampleSource"] == 4
-#         assert not "sw.tracestate_parent_id" in span_server.attributes
-#         assert not "SWKeys" in span_server.attributes
-#
-#         # Check outgoing request span tracestate has no `sw` key
-#         # because no valid parent context
-#         expected_trace_state = trace_api.TraceState([])
-#         assert span_client.context.trace_state.get("sw") == expected_trace_state.get("sw")  # None
-#
-#         # Check outgoing request span attributes
-#         #   :absent:
-#         #     service entry internal KVs, which are only on entry spans
-#         #     sw.tracestate_parent_id, because cannot be set without attributes at decision
-#         #     SWKeys, because no xtraceoptions in otel context
-#         assert not any(attr_key in span_client.attributes for attr_key in self.SW_SETTINGS_KEYS)
-#         assert not "sw.tracestate_parent_id" in span_client.attributes
-#         assert not "SWKeys" in span_client.attributes
-#
-#         # Check span_id of the outgoing request span (client span) matches
-#         # the span_id portion in the outgoing tracestate header, which
-#         # is stored in the test app's response body (new_span_id).
-#         # Note: context.span_id needs a 16-byte hex conversion first.
-#         assert "{:016x}".format(span_client.context.span_id) == new_span_id
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+import re
+import json
+import time
+from unittest import mock
+
+from opentelemetry import trace as trace_api
+
+from .test_base_sw_headers_attrs import TestBaseSwHeadersAndAttributes
+
+class TestScenario1(TestBaseSwHeadersAndAttributes):
+    """
+    Test class for starting a new tracing decision with no input headers.
+    """
+
+    def test_scenario_1_sampled(self):
+        """
+        Scenario #1, sampled:
+        1. Decision to sample is made at root/service entry span (mocked). There is no
+           OTel context extracted from request headers, so this is the root and start
+           of the trace.
+        2. Headers in the original request are not altered by the SW propagator.
+        3. Some traceparent and tracestate are injected into service's outgoing request
+           (done by OTel TraceContextTextMapPropagator).
+        4. Sampling-related attributes are set for the root/service entry span.
+        5. The span_id of the outgoing request span matches the span_id portion in the
+           tracestate header.
+        """
+        # Use in-process test app client and mock to propagate context
+        # and create in-memory trace
+        resp = None
+        # Mock JSON read to guarantee sample decision
+        timestamp = time.time()
```

Comment:
need integer casting. e.g. timestamp = int(time.time())

Reply:
Addressed in [645909a](https://github.com/solarwinds/apm-python/pull/605/commits/645909a8b9b1db9129409c7d01e825a99908723f)